### PR TITLE
Implement enhancement described in issue #27

### PR DESCRIPTION
**Pull request by eleidan**
Update App.Views.VacationRequestsList so that it is possible to cancel
any vacation with status other than `cancelled`.
Also, search filter is enabled. This functionality is provided
by the plugin out of box. For details on search filter see
http://bootstrap-table.wenzhixin.net.cn/documentation/#table-options

The search filter is supposed to replace previous filter mechanism
implementation.

@alazarchuk , @epmlys , @rubycop , @afurm

### CODE REVIEW

File: app/assets/javascripts/views/vacation_requests_list.js [hunk 1]
```diff
@@ -33,62 +27,64 @@ App.Views.VacationRequestsList = Backbone.View.extend({
...
-    var dateFrom  = this.dateToMS(this.dateFrom);
-    var dateTo    = this.dateToMS(this.dateTo);
-    var modelFrom = this.dateToMS(model.get('start_date'));
-
-    var hasProperStatus = _.include(this.statusFilter, model.get('status'));
-    var hasProperDateRange = (modelFrom >= dateFrom) && ((modelFrom <= dateTo) || (dateTo === 0));
-
-    isVisible = hasProperStatus && hasProperDateRange;
+  ownerOperationsFormatter: function(value, row, index) {
+    if (row.status === 'cancelled') {
```
Comment (afurm) [inline app/assets/javascripts/views/vacation_requests_list.js]: Is it Ok to have statuses hardcoded? Why not just define all valid statuses somewhere. In that case you can change status in one place and it apply everywhere. Does it make sense?

Comment (eleidan) [inline app/assets/javascripts/views/vacation_requests_list.js]: Well, I thought about that. You see, I'd like to have a clue what is going on with statuses. If I am forced to store statuses in an array, I have to maintain all the changes though out the code anyway, ain't I? And, I have to memorize which status does particular index represent.

Let's see:
```JS
if (row.status === 'cancelled')
```
vs
```JS
if (row.status === App.statuses[3])
```

Which one is better?

I prefer to hardcode statuses. I just don't know how to better handle this problem in JavaScript.

Comment (afurm) [inline app/assets/javascripts/views/vacation_requests_list.js]: Can it be `App.statuses.cancelled` ?

Comment (eleidan) [inline app/assets/javascripts/views/vacation_requests_list.js]: Looks OK.
To be implemented.

Comment (eleidan) [inline app/assets/javascripts/views/vacation_requests_list.js]: Done.

**Comment by afurm** [comment]
Looks good now,  thanks :+1: 

I believe there are a lot of places where we still use statuses as hardcoded values,  why we can't replace old statuses with a new model. What do you think?

**Comment by eleidan** [comment]
I think it should be done in scope of the enhancement #38

**Pull request merged**